docs/src/understand/configure-federation.md

```diff
@@ -395,7 +395,166 @@ cargohold:
       federationDomain: example.com # your chosen "backend domain"
 ```
 
-### Configure federator process to run and allow incoming traffic 
+(configure-federation-strategy-in-brig)=
+
+### Configure federation strategy (whom to federate with) in brig
+
+**Since [PR#3260](https://github.com/wireapp/wire-server/pull/3260).**
+
+You also need to define the federation strategy (whom to federate
+with), and the frequency with which the other backend services will
+refresh their cache of this configuration.
+
+``` yaml
+# override values for wire-server
+# (e.g. under ./helm_vars/wire-server/values.yaml)
+brig:
+  config:
+    optSettings:
+      setFederationStrategy: allowNone # [allowAll | allowDynamic | allowNone]
+      setFederationDomainConfigsUpdateFreq: 10 # seconds
+```
+
+The default strategy of `allowNone` effectively disables federation
+(and probably isn't what you want if you are reading this).
+`allowAll` federates with any backend that requests contact or that a
+user uses in a search.  `allowDynamic` only federates with known
+remote backends listed in cassandra.
+
+The update frequence determines how often other services will refresh
```

Comment:
I understand that having a single source of truth (`brig` in this case) guards against federation configs in other services going out of sync.

However, I'm wondering if this has to be solved with *polling* by the depending services? Can't `brig` tell the other services when its configuration changed by querying an endpoint (e.g. `POST galley/i/federation-config-changed`)? Initial sync could happen when a pod starts.

I haven't read the code, so I might be wrong here.

My impression is that these settings change very seldom and that wouldn't justify querying an endpoint every 10 Seconds or so. :thinking:

Reply:
you're right, that would work; the question is is it worth the trouble?  what is worse: running a few 100 bogus requests inside wire-server every 10 secs, or implementing more internal end-points and clients that use it?

see paragraph below on traffic (line 450).